packages/grafana-data/src/vector/ArrayVector.test.ts

```diff
@@ -0,0 +1,39 @@
+import { Field, FieldType } from '../types';
+
+import { ArrayVector } from './ArrayVector';
+
+describe('ArrayVector', () => {
+  it('should init 150k with 65k Array.push() chonking', () => {
+    const arr = Array.from({ length: 150e3 }, (v, i) => i);
+    const av = new ArrayVector(arr);
+
+    expect(av.toArray()).toEqual(arr);
+  });
+
+  it('should support add and push', () => {
+    const av = new ArrayVector<number>();
+    av.add(1);
+    av.push(2);
+    av.push(3, 4);
+
+    expect(av.toArray()).toEqual([1, 2, 3, 4]);
+  });
+
+  it('typescript does not complain about nested types', () => {
```

Comment:
Why not?
`Field<number | undefined | null | string>`

Reply:
I updated the test and comment to better reflect what is happening.  In general we are looking for types that will let existing Vector code work without change -- even when less than totally accurate! since we really want people to move to simple `T[]` -- somehow the collapsed interface now lets typescript be smarter so it says things like:

```
field.values = new ArrayVector([]);
```

```
Type 'ArrayVector<number | undefined>' is not assignable to type 'Vector<number>'.
```
even though empty is totally reasonable (and this code exists all over our tests)

Reply:
ah, now I understand.